wazo_confd/plugins/user/resource.py

```diff
@@ -61,3 +74,72 @@ def put(self, id):
     @required_acl('confd.users.{id}.delete')
     def delete(self, id):
         return super().delete(id)
+
+
+class UserListV2(ListResource):
+
+    api_version = '2.0'
+
+    def __init__(
+        self,
+        user_service,
+        line_service,
+        extension_service,
+        extension_line_service,
+        line_endpoint_sip_association_service,
+        endpoint_sip_service,
+        user_line_service,
+        wazo_user_service,
+        line_dao,
+        sip_dao,
+        user_dao,
+        transport_dao,
+    ):
+        self._user_list_resource = UserList(user_service, json_path='user')
+        self._line_list_resource = LineList(line_service)
+        self._user_line_item_resource = UserLineItem(user_line_service, user_dao, line_dao)
+        self._extension_line_list_resource = LineExtensionList(extension_line_service, extension_service, line_dao)
+        self._endpoint_sip_list_resource = SipList(endpoint_sip_service, sip_dao, transport_dao)
+        self._line_endpoint_sip_association_resource = LineEndpointAssociationSip(line_endpoint_sip_association_service, line_dao, sip_dao)
+        self._line_list_resource.schema = LineSchemaV2
+        self._wazo_user_service = wazo_user_service
+
+    def build_headers(self, user):
+        return {'Location': url_for('users', id=user['id'], _external=True)}
+
+    @required_acl('confd.users.create')
+    def post(self):
+        body = UserSchemaV2().load(request.get_json())
+
+        logger.info("Create User resource")
+        user_dict, _, _ = self._user_list_resource.post()
+        line_list = []
+
+        for line_body in body.get('lines') or []:
+            extensions = line_body.pop('extensions', None) or []
+            endpoint_sip = line_body.pop('endpoint_sip')
+            endpoint_sip, _, _ = self._endpoint_sip_list_resource._post(endpoint_sip)
+            line, _, _ = self._line_list_resource._post(line_body)
+            self._line_endpoint_sip_association_resource.put(line['id'], endpoint_sip['uuid'])
+            line['endpoint_sip'] = endpoint_sip
+            line['extensions'] = []
+            for extension_body in extensions:
+                extension, _, _ = self._extension_line_list_resource._post(line['id'], extension_body)
```

Comment:
What happens if the extension creation fails? Can you confirm that nothing is committed in database?